x/programs/runtime/import_program.go

```diff
@@ -49,17 +51,21 @@ func NewProgramModule(r *WasmRuntime) *ImportModule {
 				newInfo.Params = input.Params
 				newInfo.Fuel = input.Fuel
 
-				result := r.CallProgramToResponse(
+				result, err := r.CallProgram(
 					context.Background(),
 					&newInfo)
+				if err != nil {
+
+					return Err[RawBytes, ErrorCode](ExecutionFailure), nil
+				}
```

Comment:
Don't we want the behaviour to match with program2program calls ? Here it's returning an `ExecutionFailure` in every case.

Reply:
edit: nice you've done it at the next commit